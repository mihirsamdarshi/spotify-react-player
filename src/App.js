import React, { Component } from 'react'

import './App.css'

import AppHeader from './AppHeader'
import SearchForm from './SearchForm'

import { apiHost } from './api'

class App extends Component {
  // Because App is the "uppermost" component (see index.js), code in its componentDidMount
  // method is equivalent to an overall initialization routine. Note however that every component
  // can have its own componentDidMount, and so initialization can be separated on a per-component
  // basis.
  componentDidMount() {
    apiHost('http://api.giphy.com/v1/')
  }

  // The render method is the heart of every React component: it returns the componentâ€™s content
  // in terms of HTML and other components. It must return one and only one top-level element.
  render() {
    return (
      <div className="App">
        <AppHeader />
        <SearchForm />
      </div>
    )
  }
}

export default App
