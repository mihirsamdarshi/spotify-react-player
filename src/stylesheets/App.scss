.app-body {
  margin-top: 20px;
}

.spaceMainContent {
  margin-bottom: 15px;
}