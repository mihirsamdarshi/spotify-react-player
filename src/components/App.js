import React, {Component} from 'react';

import '../stylesheets/App.scss';

import AppHeader from './AppHeader';
import MainAppBody from './MainAppBody';

import {apiHost} from '../scripts/api';

class App extends Component {
    // Because App is the "uppermost" component (see index.js), code in its componentDidMount
    // method is equivalent to an overall initialization routine. Note however that every component
    // can have its own componentDidMount, and so initialization can be separated on a per-component
    // basis.
    componentDidMount() {
        apiHost('http://api.giphy.com/v1/');
    }

    // The render method is the heart of every React component: it returns the componentâ€™s content
    // in terms of HTML and other components. It must return one and only one top-level element.
    render() {
        return (
            <div className="App">
                <AppHeader />
                <div className="app-body">
                    <MainAppBody/>
                </div>
            </div>
        );
    }
}

export default App;
