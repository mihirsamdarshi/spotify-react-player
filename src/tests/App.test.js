import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import { mount } from 'enzyme';
import App from '../components/App';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('matches the snapshot', async () => {
    await ReactTestUtils.act(async () => {
        const component = await mount(<App/>);
        expect(component).toMatchSnapshot();
    });
});

describe('the App', () => {
    let div;
    beforeEach(() => {
        div = document.createElement('div');
    });

    afterEach(() => {
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render Hero if no token present', async() => {
        ReactTestUtils.act(() => {
            const url = 'https://helloworld.com';
            Object.defineProperty(window, 'location', {
                value: {
                    href: url,
                    hash: '',
                },
                writable: true,
            });
        });

        await ReactTestUtils.act(async() => {
            await ReactDOM.render(<App />, div)
        });

        expect(div.querySelector('.heroInfo') !== null).toBe(true);
    });

    it('should render MainAppBody if a token is present', async() => {
        ReactTestUtils.act(() => {
            const url = '#access_token=helloworld&token_type=Bearer&expires_in=69';
            Object.defineProperty(window, 'location', {
                value: {
                    hash: url,
                },
                writable: true,
            });
        });

        await ReactTestUtils.act(async() => {
            await ReactDOM.render(<App />, div)
        });

        expect(div.querySelector('.gridContainer') !== null).toBe(true);
    });
});
