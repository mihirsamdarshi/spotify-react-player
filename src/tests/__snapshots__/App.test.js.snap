// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <AppHeader>
      <WithStyles(ForwardRef(AppBar))
        position="static"
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <ForwardRef(AppBar)
          classes={
            Object {
              "colorDefault": "MuiAppBar-colorDefault",
              "colorPrimary": "MuiAppBar-colorPrimary",
              "colorSecondary": "MuiAppBar-colorSecondary",
              "positionAbsolute": "MuiAppBar-positionAbsolute",
              "positionFixed": "MuiAppBar-positionFixed",
              "positionRelative": "MuiAppBar-positionRelative",
              "positionStatic": "MuiAppBar-positionStatic",
              "positionSticky": "MuiAppBar-positionSticky",
              "root": "MuiAppBar-root",
            }
          }
          position="static"
          style={
            Object {
              "backgroundColor": "black",
            }
          }
        >
          <WithStyles(ForwardRef(Paper))
            className="MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
            component="header"
            elevation={4}
            square={true}
            style={
              Object {
                "backgroundColor": "black",
              }
            }
          >
            <ForwardRef(Paper)
              className="MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
              classes={
                Object {
                  "elevation0": "MuiPaper-elevation0",
                  "elevation1": "MuiPaper-elevation1",
                  "elevation10": "MuiPaper-elevation10",
                  "elevation11": "MuiPaper-elevation11",
                  "elevation12": "MuiPaper-elevation12",
                  "elevation13": "MuiPaper-elevation13",
                  "elevation14": "MuiPaper-elevation14",
                  "elevation15": "MuiPaper-elevation15",
                  "elevation16": "MuiPaper-elevation16",
                  "elevation17": "MuiPaper-elevation17",
                  "elevation18": "MuiPaper-elevation18",
                  "elevation19": "MuiPaper-elevation19",
                  "elevation2": "MuiPaper-elevation2",
                  "elevation20": "MuiPaper-elevation20",
                  "elevation21": "MuiPaper-elevation21",
                  "elevation22": "MuiPaper-elevation22",
                  "elevation23": "MuiPaper-elevation23",
                  "elevation24": "MuiPaper-elevation24",
                  "elevation3": "MuiPaper-elevation3",
                  "elevation4": "MuiPaper-elevation4",
                  "elevation5": "MuiPaper-elevation5",
                  "elevation6": "MuiPaper-elevation6",
                  "elevation7": "MuiPaper-elevation7",
                  "elevation8": "MuiPaper-elevation8",
                  "elevation9": "MuiPaper-elevation9",
                  "root": "MuiPaper-root",
                  "rounded": "MuiPaper-rounded",
                }
              }
              component="header"
              elevation={4}
              square={true}
              style={
                Object {
                  "backgroundColor": "black",
                }
              }
            >
              <header
                className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
                style={
                  Object {
                    "backgroundColor": "black",
                  }
                }
              >
                <WithStyles(ForwardRef(Toolbar))>
                  <ForwardRef(Toolbar)
                    classes={
                      Object {
                        "dense": "MuiToolbar-dense",
                        "gutters": "MuiToolbar-gutters",
                        "regular": "MuiToolbar-regular",
                        "root": "MuiToolbar-root",
                      }
                    }
                  >
                    <div
                      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
                    >
                      <WithStyles(ForwardRef(Typography))
                        className="title"
                        variant="h6"
                      >
                        <ForwardRef(Typography)
                          className="title"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          variant="h6"
                        >
                          <h6
                            className="MuiTypography-root title MuiTypography-h6"
                          >
                            Applotify
                          </h6>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </div>
                  </ForwardRef(Toolbar)>
                </WithStyles(ForwardRef(Toolbar))>
              </header>
            </ForwardRef(Paper)>
          </WithStyles(ForwardRef(Paper))>
        </ForwardRef(AppBar)>
      </WithStyles(ForwardRef(AppBar))>
    </AppHeader>
    <Hero>
      <div
        className="fillWindow"
      >
        <div
          className="heroInfo"
        >
          <h1>
             Applotify 
          </h1>
          <h3>
             Play Spotify Playlists* 
          </h3>
        </div>
        <div
          className="centeredBtn"
        >
          <a
            className="spotifyLoginBtn"
            href="https://accounts.spotify.com/authorize?response_type=token&client_id=ccc1ea71746a4d4ea8468a7cc7d1d9f0&redirect_uri=http://localhost:3000&scope=user-read-private%20user-read-playback-state%20user-modify-playback-state%20streaming%20user-read-currently-playing&show_dialog=true"
          >
            Login With Spotify
          </a>
        </div>
        <p
          className="asterisk"
        >
           * only available in Chrome and Firefox browsers and to users with Spotify Premium due to Spotify API limitations
        </p>
      </div>
    </Hero>
  </div>
</App>
`;
